healthyregions/SDOHPlace
# #74, #75, #76: Change raw Solr object to local Solr object and create Parent Object for side bar

This PR is intended to resolve #74, #75 and #76. 

This PR defines the `SolrObject` interface (#74), which is the basic object type that all Solr raw objects should be transferred to before future processes in the front end. Then based on different user requirements we will transfer the SolrObject to other types of objects.

This PR also includes `SolrParent` interface (#75), which is the parent record type that is used in the sidebar of the Discovery page. Currently, they are displayed as accordions as a placeholder to show Parent Object’s data (#76), but we can update the visual components to what designers want later.


### How to Test

1. Run `npm i` to install the MUI icon library for Accordion
2. Run `npm run dev` and navigate to http://localhost:3000/search
3. You should see the parent list accordions. Expand any accordion and you should see creator information, description, and years. 
<img width="958" alt="Screenshot 2024-02-05 at 9 05 18 AM" src="https://github.com/healthyregions/SDOHPlace/assets/92752107/dfedfff6-eada-48ed-a9eb-89e914a39bf6">

Note that:
- Description of the `SolrParent` is the first sentence of the description of `SolrObject` 
- Years are generated based on child records using the `is_version_of` attribute, with *duplication removed*. If we want to see duplicate years, I can change the Set back to Array for the `years` attribute of the `SolrParent` interface.
- **When defining the parent object, the basic idea is that as long as the object has no `is_version_of`, this object is a parent object even though there could be no child**. If we require a parent object **must** has years (i.e. children), then I'll update the code accordingly
